File: skellig-test-step-processing-tcp/src/main/java/org/skellig/teststep/processor/tcp/model/TcpDetails.java
package org.skellig.teststep.processor.tcp.model;

public class TcpDetails {

    private String channelId;
    private String hostName;
    private int port;

    public TcpDetails(String channelId, String hostName, int port) {
        this.channelId = channelId;
        this.hostName = hostName;
        this.port = port;
    }

    public String getChannelId() {
        return channelId;
    }

    public String getHostName() {
        return hostName;
    }

    public int getPort() {
        return port;
    }

}

File: skellig-test-step-processing-tcp/src/main/kotlin/org/skellig/teststep/processor/tcp/model/TcpDetails.kt
package org.skellig.teststep.processor.tcp.model

/**
 * Represents the details of a TCP connection, including the ID, host name, port, and whether the connection should
 * keep alive.
 *
 * @property id The unique identifier for the TCP connection which represents a channel ID and used in test step to
 * indicate which TCP channel to communicate with.
 * @property hostName The host name or IP address of the TCP server.
 * @property port The port number of the TCP server.
 * @property isKeepAlive Indicates whether the TCP connection should be kept alive.
 */
class TcpDetails(
    val id: String,
    val hostName: String,
    val port: Int,
    val isKeepAlive: Boolean = true
) {

    override fun toString(): String {
        return "$hostName:$port (isKeepAlive = $isKeepAlive)"
    }
}